File: ThirtyFour.UI/ThirtyFour.UI.Tests/Table/ChildElementCellParserTests.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;
using OpenQA.Selenium;
using ThirtyFour.UI.Table;

namespace ThirtyFour.UI.Tests.Tests.Table
{
    [TestClass]
    public class ChildElementCellParserTests : BaseTestSuite
    {
        
        [TestMethod]
        public void ParserReturnsCorrectElement()
        {
            IWebElement cell = driver.FindElement(By.Id("first-action-cell"));
            ICellParser<IWebElement> parser = new ChildElementCellParser(By.TagName("button"));
            Assert.AreEqual("first-button", parser.GetValue(cell).GetAttribute("id"));
        }
    }
}

File: java/src/test/java/com/github/danhumphrey/thirtyfour/ui/test/table/ChildElementCellParserTests.java
package com.github.danhumphrey.thirtyfour.ui.test.table;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.core.IsEqual.equalTo;

import org.junit.jupiter.api.Test;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;

import com.github.danhumphrey.thirtyfour.ui.table.ChildElementCellParser;
import com.github.danhumphrey.thirtyfour.ui.test.BaseTestSuite;

public class ChildElementCellParserTests extends BaseTestSuite {
	@Test
	public void testParserReturnsCorrectElement() {
		WebDriver driver = getDriver();
		WebElement cell = driver.findElement(By.id("first-action-cell"));
		ChildElementCellParser parser = new ChildElementCellParser(By.tagName("button"));
		assertThat(parser.getValue(cell).getAttribute("id"), equalTo("first-button") );
	} 
}
